implementations/rust/ockam/ockam_command/src/util/mod.rs

```diff
@@ -534,3 +539,24 @@ pub async fn query_pid(
     tx.send(Some(status.pid)).unwrap();
     Ok(())
 }
+
+#[test]
+fn test_extract_node_name() {
+    let test_cases = vec![
+        ("/node/test", "test", false),
+        ("test", "test", false),
+        ("/test", "/test", false),
+        ("/node/", "", false),
+        ("/node/test/tcp", "", true),
+        ("/node/test/test", "", true),
+        ("/node/test/tcp/22", "test", false),
+    ];
+
+    for (input, check, fails) in test_cases {
+        if !fails {
+            assert!(extract_node_name(input).unwrap() == check);
+        } else {
+            assert!(extract_node_name(input).is_err());
+        }
+    }
```

Comment:
You can simplify this if you define the test cases with `Result`'s directly:

```rust
let test_cases = vec![
    ("/node/test", Ok("test")),
    ("/node/test/test", Err(())),
    ...
];
```

Then here you can just 

```rust
for (input, res) in test_cases {
    if let Ok(n) = res {
        assert!(extract_node_name(input).unwrap() == n);
    } else {
        assert!(extract_node_name(input).is_err());
    }
}
```